### PR TITLE
bug:TriggersAnnotationEnricher should not execute in PlatformMode.kubernetes 

### LINKED ISSUE #2559: TriggersAnnotationEnricher should not execute in `PlatformMode.kubernetes`
### Component

JKube Kit

### Description

According to [Red Hat OpenShift Triggering updates on ImageStream documentation](https://docs.openshift.com/container-platform/4.8/openshift_images/triggering-updates-on-imagestream-changes.html), annotation `image.openshift.io/triggers` is defined to be able to trigger updates in OpenShift Container Platform.

We have [TriggersAnnotationEnricher](https://github.com/eclipse/jkube/blob/master/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricher.java) responsible for adding this annotation. Currently, it seems to be executed in all cases. Since it's a Red Hat OpenShift Container Platform related annotation, it should be executed only when `PlatformMode.openshift` is specified.

We should add a check to return when `platformMode != PlatformMode.openshift`

https://github.com/eclipse/jkube/blob/c387c2be37561f4a07788de0ce3aa63ed2a8b4fc/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricher.java#L74

### Expected Behavior
TriggersAnnotationEnricher should not execute in `PlatformMode.kubernetes`

### Acceptance Criteria
- [x] Add a check for only executing the `enrich` method when `platformMode == PlatformMode.openshift`
- [x] Add a unit test to validate the abovementioned condition
- [x] Update integration tests which have references to `image.openshift.io/triggers` annotation in `kubernetes.yml`

### PR DESCRIPTION
**Pull request by AmitPatil05**
## Description

Fixes #2559

Added a check in the enriching process to only do it if the Kubernetes environment is OpenShift. Also added 2 new unit tests and updated releevant unit tests.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] My code follows the style guidelines of this project
 - [ ] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [ ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [ ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricherTest.java [hunk 1]
```diff
@@ -55,6 +56,34 @@ void setUp() {
         context = mock(JKubeEnricherContext.class,RETURNS_DEEP_STUBS);
     }
 
+    @Test
+    void testEnrichmentNotPerformedInNonOpenShiftMode() throws IOException {
+        // Create a mock KubernetesListBuilder
```
[the remark this answers is not in the document]
Comment (rohanKanojia) [reply jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricherTest.java:61]: If you want to add a description maybe use `@DisplayName` annotation

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricherTest.java [hunk 2]
```diff
@@ -55,6 +56,34 @@ void setUp() {
...
+        // Call the enrich method with PlatformMode.kubernetes
+        enricher.enrich(PlatformMode.kubernetes, builder);
+
+        // Verify that the accept method is not executed
+        verify(builder, never()).accept(any());
+    }
+    @Test
+    void testEnrichmentPerformedInOpenShiftMode() throws IOException {
+        // Create a mock KubernetesListBuilder
+        KubernetesListBuilder builder = mock(KubernetesListBuilder.class);
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricherTest.java:76]: Could you please adapt these tests like existing tests in this class? You're using way too much mocking.

Try to create a real KubernetesListBuilder with some valid Kubernetes object. For this case verify that the object contains the annotation after `enricher.enrich` gets called. For `testEnrichmentNotPerformedInNonOpenShiftMode` you can verify that object does not contain that annotation

**Comment by AmitPatil05** [comment]
Hello, 
As per your comment, I have created a real KubernetesListBuilder with a valid Kubernetes object (StatefulSet).

1. In testEnrichmentNotPerformedInNonOpenShiftMode case, I called enrich method with PlatformMode.kubernetes. Verification was done in such a way that, the StatefulSet is retrieved from the builder, and assertions are made to verify the presence of the original annotation ("annkey": "annvalue") and the absence of any additional annotations, ensuring that only the original annotation remains unchanged by using hasSize(1).
2. In testEnrichmentPerformedInOpenShiftMode case, I called enrich method with PlatformMode.openshift. Verification was done in such a way that, the original annotation stays ("annkey": "annvalue") and the addition of a new annotation ("image.openshift.io/triggers") related to OpenShift triggers is done. Ensured that the total number of annotations being 2 using hasSize(2), which indicates that both the original and new annotations are present.


Can you please review and tell if this approach is fine? 


Thanks,
Amit

**Comment by AmitPatil05** [comment]
Sonar build is failing because of [INFO] Kubernetes Maven Plugin :: Integration Tests ....... FAILURE [04:17 min]

I am not able to find the root cause of this. The change is just in test. Maybe is it because of the change in PlatformMode to openshift in the test class? Also it might be because of now enricher is specifically using openshift as platformMode and not anything else. 

Can you please guide me here?

**Comment by rohanKanojia** [comment]
@AmitPatil05 : Your changes now don't execute in kubernetes mode. Some integration testshave wrong expectations. You need to remove this annotation from `expected/kubernetes.yml` 

https://github.com/eclipse/jkube/blob/62080bfd0ce8c65245fef69723c95816d49661bb/kubernetes-maven-plugin/it/src/it/dependency-resources/expected/kubernetes.yml#L75

**Comment by rohanKanojia** [comment]
@AmitPatil05 : Kubernetes Integration tests seem to be fixed now.

However, I see some occurrences of this annotation in `gradle-plugin/it` module as well:
```
src/it/controller/expected/daemonset/kubernetes.yml:12:        image.openshift.io/triggers: "@ignore@"
src/it/controller/expected/replicaset/kubernetes.yml:12:        image.openshift.io/triggers: "@ignore@"
src/it/controller/expected/statefulset/kubernetes.yml:12:        image.openshift.io/triggers: "@ignore@"
src/it/triggersannotation/expected/kubernetes.yml:12:      image.openshift.io/triggers: |-
```

**Comment by AmitPatil05** [comment]
Hi @rohanKanojia 
I removed all the annotation from kubernetes.yaml files in 7da3367 file. The Sonar build failed. Then I put back the annotations in some of the files in the latest commit. Still the Sonar build failed. Now the Kubernetes gradle plugin build is successful but integrationn test is now failing. Does this means some of the unit tests are failing right now? Is there something wrong with what I implemented in the test?

File: gradle-plugin/it/src/it/triggersannotation/expected/kubernetes.yml [hunk 3]
```diff
@@ -9,7 +9,6 @@ items:
       jkube.eclipse.org/git-url: "@ignore@"
       jkube.eclipse.org/git-commit: "@ignore@"
       jkube.eclipse.org/git-branch: "@ignore@"
-      image.openshift.io/triggers: |-
```
Comment (rohanKanojia) [inline gradle-plugin/it/src/it/triggersannotation/expected/kubernetes.yml:12]: @AmitPatil05 You haven't removed this correctly. This attribute has multi line scalar value

You need to delete till line 19 (where json array ends)

**Comment by AmitPatil05** [comment]
Hi @rohanKanojia , I removed the annotations correctly. Can you please review the latest changes?

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricherTest.java [hunk 4]
```diff
@@ -55,6 +55,56 @@ void setUp() {
...
+                    .endTemplate()
+                .endSpec()
+                .build());
+
+        TriggersAnnotationEnricher enricher = new TriggersAnnotationEnricher(context);
+        enricher.enrich(PlatformMode.kubernetes, builder);
+
+        StatefulSet res = (StatefulSet) builder.build().getItems().get(0);
+        assertThat(res.getMetadata().getAnnotations())
+            .containsEntry("annkey", "annvalue")
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricherTest.java:79]:
```suggestion
            .containsEntry("annkey", "annvalue")
            .doesNotContainKey("image.openshift.io/triggers")
```

**Comment by rohanKanojia** [comment]
@AmitPatil05 : Please squash your commits into one commit.

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**